model/Shop/ShopUbytovani.php

```diff
@@ -319,7 +354,9 @@ public function ubytovaniHtml(bool $muzeEditovatUkoncenyProdej = false)
         $this->htmlDny($t, $muzeEditovatUkoncenyProdej);
         // sloupce popisků
         foreach ($this->mozneTypy as $typ => $predmet) {
-            if ($typ != 'Spacák' && !$this->maPravoNaPokoj($this->uzivatel())){
+
+            if ($this->jeOmezeniNaSpacaky() && $typ != 'Spacák' &&
```

Comment:
Tady v každé iteraci cyklu znovu a znovu zjišťujeme, zda účastník má právo na postel.

Optimálnější je zjistit si to před cyklem jednou, hodit do lokální proměnné a čerpat už jenom z ní.